pkg/controllers/static/deprovisioning/controller.go

```diff
@@ -159,3 +169,69 @@ func (c *Controller) Register(_ context.Context, m manager.Manager) error {
 func HasNodePoolReplicaCountChanged(oldNP, newNP *v1.NodePool) bool {
 	return lo.FromPtr(oldNP.Spec.Replicas) != lo.FromPtr(newNP.Spec.Replicas)
 }
+
+// Returns nodes suitable for deprovisioning, prioritizing:
+// 1. Empty nodes (nodes with no pods or only DaemonSet pods without do-not-disrupt annotation)
+// 2. If more nodes needed, nodes with lowest disruption cost (nodes with pods that have do-not-disrupt will have highest cost)
+func (c *Controller) getDeprovisioningCandidates(ctx context.Context, np *v1.NodePool, nodes []*state.StateNode, count int) []*state.StateNode {
+	// First get empty nodes
+	emptyNodes := lo.Filter(nodes, func(node *state.StateNode, _ int) bool {
+		pods, err := node.Pods(ctx, c.kubeClient)
+		if err != nil {
+			log.FromContext(ctx).WithValues("node", node.Name()).Error(err, "unable to list pods, treating as non-empty")
+			return false
+		}
+		return len(pods) == 0 || lo.EveryBy(pods, pod.IsOwnedByDaemonSet) && lo.NoneBy(pods, pod.HasDoNotDisrupt)
```

Comment:
What was the rationale for deviating from the emptiness check logic that we have in the disruption controller? Is there a reason that this line had to specifically be different?